Proxy/proxy.py

```diff
@@ -351,10 +362,106 @@ def callsign2COM():
     return json.loads(local)
 
 
+def initDB():
+    """
+    Initialize database, creates it if not present
+
+    :return: True or False if successful
+    """
+
+    # Obtain configuration file names
+    try:
+        dbFilename = proxyConfig.get("DATABASE", "FILENAME")
+        dbSchema = proxyConfig.get("DATABASE", "SCHEMANAME")
+
+    except ConfigParser.Error as e:
+        logger.error("ConfigParse.Error: " + str(e))
+        return False
+
+    # Check if database exists
+    if os.path.isfile(dbFilename):
+        pass
+    else:
+        # Open database schema SQL file and execute the SQL functions inside
+        # after connecting. Close the database when complete.
+        try:
+            with open(dbSchema, 'rt') as f:
+                conn = sqlite3.connect(dbFilename)
+                cur = conn.cursor()
+                schema = f.read()
+                cur.executescript(schema)
+            conn.close()
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            return False
+
+    return True
+
+
+def sqlInsert(data):
+    """
+    Takes in a data tuple and inserts int into the Proxy SQLite table
+
+    :param data: Proxy dictionary
+    :return: Status True or False on SQL insertion success
+    """
+
+    # Read in name of database
+    try:
+        db = proxyConfig.get("DATABASE", "FILENAME")
+
+    except ConfigParser.Error as e:
+        logger.error("ConfigParse.Error: " + str(e))
+        return False
+
+    # # Change dictionary into list with proper order
+    sqlparameters = [None, data["port"],data["data"],time.time()]
+
+    if len(sqlparameters) > 0:
+        # Create parameter substitute "?" string for SQL query then create SQL
+        numKeys = len(sqlparameters)
+        paramSubs = "?" * (numKeys)
+        paramSubs = ",".join(paramSubs)
+        sql = "INSERT INTO PROXY VALUES(" + paramSubs + ")"
+
+        # Connect to database, create SQL query, execute query, and close database
+        try:
+            conn = sqlite3.connect(db)
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            return False
+
+        # Connect to database, create SQL query, execute query, and close database
+        try:
+            conn = sqlite3.connect(db)
+            # Use connection as context manager to rollback automatically if error
+            with conn:
+                conn.execute(sql,sqlparameters)
+
+        except sqlite3.Error as e:
+            logger.error("Sqlite3.Error: " + str(e))
+            conn.rollback()
```

Comment:
Using `conn` as a context manager will automatically call `rollback()` so it is not necessary here.